drivers/mipi_dsi/dsi_stm32.c

```diff
@@ -0,0 +1,500 @@
+	LOG_DBG("  LPHorizontalBackPorchEnable 0x%x", data->vid_cfg.LPHorizontalBackPorchEnable);
+	LOG_DBG("  LPVerticalActiveEnable 0x%x", data->vid_cfg.LPVerticalActiveEnable);
+	LOG_DBG("  LPVerticalFrontPorchEnable 0x%x", data->vid_cfg.LPVerticalFrontPorchEnable);
+	LOG_DBG("  LPVerticalBackPorchEnable 0x%x", data->vid_cfg.LPVerticalBackPorchEnable);
+	LOG_DBG("  LPVerticalSyncActiveEnable 0x%x", data->vid_cfg.LPVerticalSyncActiveEnable);
+	LOG_DBG("  FrameBTAAcknowledgeEnable 0x%x", data->vid_cfg.FrameBTAAcknowledgeEnable);
+
+	if (config->active_errors) {
+		LOG_DBG("HAL_DSI_ConfigErrorMonitor: 0x%x", config->active_errors);
+	}
+
+	if (config->lp_rx_filter_freq) {
+		LOG_DBG("HAL_DSI_SetLowPowerRXFilter: %d", config->lp_rx_filter_freq);
+	}
+
+	if (data->host_timeouts) {
+		DSI_HOST_TimeoutTypeDef *ht = data->host_timeouts;
+
+		LOG_DBG("HAL_DSI_ConfigHostTimeouts:");
+		LOG_DBG("  TimeoutCkdiv %u", ht->TimeoutCkdiv);
+		LOG_DBG("  HighSpeedTransmissionTimeout %u", ht->HighSpeedTransmissionTimeout);
+		LOG_DBG("  LowPowerReceptionTimeout %u", ht->LowPowerReceptionTimeout);
+		LOG_DBG("  HighSpeedReadTimeout %u", ht->HighSpeedReadTimeout);
+		LOG_DBG("  LowPowerReadTimeout %u", ht->LowPowerReadTimeout);
+		LOG_DBG("  HighSpeedWriteTimeout %u", ht->HighSpeedWriteTimeout);
+		LOG_DBG("  HighSpeedWritePrespMode %u", ht->HighSpeedWritePrespMode);
+		LOG_DBG("  LowPowerWriteTimeout %u", ht->LowPowerWriteTimeout);
+		LOG_DBG("  BTATimeout %u", ht->BTATimeout);
+	}
+
+	if (data->phy_timings) {
+		LOG_DBG("HAL_DSI_ConfigPhyTimer:");
+		LOG_DBG("  ClockLaneHS2LPTime %u", data->phy_timings->ClockLaneHS2LPTime);
+		LOG_DBG("  ClockLaneLP2HSTime %u", data->phy_timings->ClockLaneLP2HSTime);
+		LOG_DBG("  DataLaneHS2LPTime %u", data->phy_timings->DataLaneHS2LPTime);
+		LOG_DBG("  DataLaneLP2HSTime %u", data->phy_timings->DataLaneLP2HSTime);
+		LOG_DBG("  DataLaneMaxReadTime %u", data->phy_timings->DataLaneMaxReadTime);
+		LOG_DBG("  StopWaitTime %u", data->phy_timings->StopWaitTime);
+	}
+}
+
+static int mipi_dsi_stm32_host_init(const struct device *dev)
+{
+	const struct mipi_dsi_stm32_config *config = dev->config;
+	struct mipi_dsi_stm32_data *data = dev->data;
+	struct stm32_pclken pclken_tmp;
+	uint32_t hse_clock;
+	int ret;
+
+	switch (config->data_lanes) {
+	case 1:
+		data->hdsi.Init.NumberOfLanes = DSI_ONE_DATA_LANE;
+		break;
+	case 2:
+		data->hdsi.Init.NumberOfLanes = DSI_TWO_DATA_LANES;
+		break;
+	default:
+		LOG_ERR("Number of DSI lanes (%d) not supported!", config->data_lanes);
+		return -ENOTSUP;
+	}
+
+	pclken_tmp.bus = STM32_SRC_PLL3_R;
+	ret = clock_control_get_rate(config->rcc, &pclken_tmp, &data->pixel_clk_khz);
+	if (ret) {
+		LOG_ERR("Get pixel clock failed! (%d)", ret);
+		return ret;
+	}
+
+	data->pixel_clk_khz /= 1000;
+	pclken_tmp.bus = STM32_SRC_HSE; /* external resonator as clock source is mandatory */
+	ret = clock_control_get_rate(config->rcc, &pclken_tmp, &hse_clock);
```

Comment:
Please add a dedicated clock.
CF STM32 eth driver to see how to describe multiple clocks from DT